[Baikal] Update to 0.12.1 fails on composer install (lock file mismatch)

### ✅ Have you read and understood the above guidelines?

yes, I scanned and only found the 1 that was related.

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

Baikal

### 📂 What was the exact command used to execute the script?

update (ssh)

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 🧱 Is this Proxmox host running arm64?

No

### 📈 Which Proxmox version are you on?

8.4.20

### 📝 Provide a clear and concise description of the issue.

Updating Baikal 0.11.1 to 0.12.1 fails during composer install because the composer.lock bundled in the 0.12.1 release doesn't satisfy that release's own composer.json (sabre/dav, twig, symfony/yaml version mismatches, plus twbs/bootstrap, fortawesome/font-awesome, components/jquery, and phpstan-deprecation-rules missing from the lock entirely).

This looks like an upstream packaging issue in Baikal's 0.12.1 tag rather than a script bug, possibly related to that release's "reduce space consumption of release zip" change.

Related but distinct from #14693 (that was a fresh install with no lock file at all), this is an update where the lock exists but doesn't match.

### 🔄 Steps to reproduce the issue.

1. Run the Baikal update from 0.11.1 to 0.12.1 via the script.

### ❌ Paste the full error output (if available).

This is where it failed.

Module php8.3 already enabled
  ✔️  Setup PHP 8.3
  ⏳  Update Composer 2.9.5Upgrading to version 2.10.2 (stable channel).
   
Use composer self-update --rollback to return to version 2.9.5
  ✔️  Update Composer 2.10.2
  ✔️  Deployed: baikal (0.12.1)
  ✔️  Restored data
  ⏳  Configuring BaikalInstalling dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
- Required package "sabre/dav" is in the lock file as "4.7.0" but that does not satisfy your constraint "~4.7.1".
- Required package "twig/twig" is in the lock file as "v3.22.2" but that does not satisfy your constraint "~3.27.1".
- Required package "symfony/yaml" is in the lock file as "v7.3.8" but that does not satisfy your constraint "~7.4.8".
- Required package "twbs/bootstrap" is not present in the lock file.
- Required package "fortawesome/font-awesome" is not present in the lock file.
- Required package "components/jquery" is not present in the lock file.
- Required (in require-dev) package "phpstan/phpstan" is in the lock file as "2.1.40" but that does not satisfy your constraint "2.1.53".
- Required (in require-dev) package "phpstan/phpstan-deprecation-rules" is not present in the lock file.
This usually happens when composer files are incorrectly merged or the composer.json file is manually edited.
Read more about correctly resolving merge conflicts https://getcomposer.org/doc/articles/resolving-merge-conflicts.md
and prefer using the "require" command over editing the composer.json file directly https://getcomposer.org/doc/03-cli.md#require-r
  ✖️  in line 49: exit code 4 (Command failed with exit status 4): while executing command  composer install
  --- Last 20 lines of log ---
[2026-08-16 09:57:50] [OK] Update available: baikal 0.11.1 → 0.12.1
[2026-08-16 09:57:50] [INFO] Stopping Service
[2026-08-16 09:57:50] [OK] Stopped Service
[2026-08-16 09:57:50] [INFO] Backing up data
[2026-08-16 09:57:50] [OK] Backed up data to /opt/baikal.backup
(Reading database ... 32075 files and directories currently installed.)
Preparing to unpack .../debsuryorg-archive-keyring.deb ...
Unpacking debsuryorg-archive-keyring (2025.11.18) over (2025.11.18) ...
Setting up debsuryorg-archive-keyring (2025.11.18) ...
[2026-08-16 09:57:53] [INFO] Installing Apache with PHP 8.3 module
[2026-08-16 09:58:09] [INFO] Installing PHP 8.3 packages
[2026-08-16 09:58:13] [OK] Setup PHP 8.3
[2026-08-16 09:58:13] [INFO] Update Composer 2.9.5
[2026-08-16 09:58:14] [OK] Update Composer 2.10.2
[2026-08-16 09:58:14] [INFO] Fetching GitHub release: baikal (0.12.1)
[2026-08-16 09:58:15] [OK] Deployed: baikal (0.12.1)
[2026-08-16 09:58:15] [INFO] Restoring data
[2026-08-16 09:58:15] [OK] Restored data
[2026-08-16 09:58:15] [INFO] Configuring Baikal
[2026-08-16 09:58:15] [ERROR] in line 49: exit code 4 (Command failed with exit status 4): while executing command  composer install
  -----------------------------------

### 🖼️ Additional context (optional).

Not sure if you'd want to report this upstream to sabre-io/Baikal or if you want me to handle it -> open to either.

Workaround that resolved it for me below.

1. Navigate to /opt/baikal
2. Run composer update --no-dev --optimize-autoloader
3. Rerun the update command and reboot (reboot might not be needed, but I performed it)

This resolved fresh compatible versions, and the services came back up properly

--- a/ct/baikal.sh
+++ b/ct/baikal.sh
@@ -46,7 +46,7 @@ function update_script() {
 
     msg_info "Configuring Baikal"
     cd /opt/baikal
-    $STD composer install
+    $STD composer install || { msg_warn "composer.lock doesn't match composer.json (known upstream packaging issue) - regenerating"; $STD composer update; }
     msg_ok "Configured Baikal"
 
     msg_info "Starting Service"

--- a/install/baikal-install.sh
+++ b/install/baikal-install.sh
@@ -25,7 +25,7 @@ PG_DB_NAME="baikal_db" PG_DB_USER="baikal_user" PG_DB_PASS="$(openssl rand -base
 
 msg_info "Configuring Baikal"
 cd /opt/baikal
-$STD composer install
+$STD composer install || { msg_warn "composer.lock doesn't match composer.json (known upstream packaging issue) - regenerating"; $STD composer update; }
 cat <<EOF >/opt/baikal/config/baikal.yaml
 database:
     backend: pgsql

